api/app/clients/classes/BaseClient.js

```diff
@@ -0,0 +1,467 @@
+const crypto = require('crypto');
+const { RecursiveCharacterTextSplitter } = require('langchain/text_splitter');
+const { ChatOpenAI } = require('langchain/chat_models/openai');
+const { loadSummarizationChain } = require('langchain/chains');
+const { refinePrompt } = require('./refinePrompt');
+const { getConvo, getMessages, saveMessage, updateMessage, saveConvo } = require('../../../models');
+
+class BaseClient {
+  constructor(apiKey, options = {}) {
+    this.apiKey = apiKey;
+    this.sender = options.sender || 'AI';
+    this.contextStrategy = null;
+    this.currentDateString = new Date().toLocaleDateString('en-us', {
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric'
+    });
+  }
+
+  setOptions() {
+    throw new Error("Method 'setOptions' must be implemented.");
+  }
+
+  getCompletion() {
+    throw new Error("Method 'getCompletion' must be implemented.");
+  }
+
+  getSaveOptions() {
+    throw new Error('Subclasses must implement getSaveOptions');
+  }
+
+  async buildMessages() {
+    throw new Error('Subclasses must implement buildMessages');
+  }
+
+  getBuildMessagesOptions() {
+    throw new Error('Subclasses must implement getBuildMessagesOptions');
+  }
```

Comment:
Just thinking here, we might be best creating some error classes for the different error types in the project

Reply:
although I get that these are yet to be implemented things